### DISCUSSION TITLE
Python support

### DISCUSSION THREAD

**Discussion by KlumpRasmus** [BelfrySCAD/BOSL2#1293] [Q&A]
Hi,

Is it possible to generate Python bindings for BOSL2 for use with this great python-enabled OpenSCAD?: http://www.guenther-sohler.net/openscad/

**Comment by Anyel-ec** [comment]
Hello my bro <3 

Yes, it is possible to generate bindings in Python for BOSL2 for the purpose of use with Python-enabled OpenSCAD. To do this, you can use the Python ctypes module, which allows you to call functions in shared libraries written in C and C++ from Python.

**Comment by revarbat** [comment]
Actually, BOSL2 is an OpenSCAD library.  How does one set things up to call OpenSCAD functions and modules, and access OpenSCAD variables (Used to implement constants like UP, DOWN, etc) from Python?

-Revar

**Reply by KlumpRasmus** [reply]
Yes, is it even possible without converting all the BOSL2 scad files into Python utilizing the Python functionality available in http://pythonscad.org/ ?

**Comment by lozdown** [comment]
Take a look at [SolidPython2](https://github.com/jeff-dh/SolidPython/wiki). Different approach to generating SCAD files and includes BOSL2
